### ISSUE TITLE
[🐛BUG] Bert4Rec性能表现非常差

### ISSUE THREAD

**Issue by zyx1017** [RUCAIBox/RecBole#1789]
**描述这个 bug**
Bert4Rec在Amazon-Beauty上的性能非常差，甚至比不过BPR、

**如何复现**
复现这个 bug 的步骤：
1. 您引入的额外 yaml 文件
BERT4REC.yaml
n_layers: 2
n_heads: 2
hidden_size: 64
inner_size: 256
hidden_dropout_prob: 0.5
attn_dropout_prob: 0.5
hidden_act: 'gelu'
layer_norm_eps: 1e-12
initializer_range: 0.02
mask_ratio: 0.6
loss_type: 'CE'

overall.yaml
# general
gpu_id: 0
use_gpu: True
seed: 2020
state: INFO
reproducibility: True
#data_path: 'dataset/'
checkpoint_dir: 'saved'
show_progress: True
save_dataset: False
dataset_save_path: ~
save_dataloaders: False
dataloaders_save_path: ~
log_wandb: False
wandb_project: 'recbole'

# directory setting
log_root: "./log/"
data_path: "./dataset/"

# training settings
epochs: 300  #300
train_batch_size: 256  #2048
learner: adam
learning_rate: 0.001
training_neg_sample_num: 0
#training_neg_sample_num: 1
#training_neg_sample_distribution: uniform
eval_step: 1
stopping_step: 20
clip_grad_norm: ~
# clip_grad_norm:  {'max_norm': 5, 'norm_type': 2}
weight_decay: 0.0
draw_loss_pic: False
require_pow: False

# evaluation settings
eval_setting: TO_LS,full #对数据按时间排序，设置留一法划分数据集，并使用全排序
group_by_user: True
split_ratio: [0.8,0.1,0.1]
leave_one_num: 2
real_time_process: False  #If True, dataloader will do data pre-processing, such as neg-sampling and data-augmentation.
metrics: ["Recall","MRR","NDCG","Precision"]
topk: [5,10,20,50]
valid_metric: MRR@10
valid_metric_bigger: True
eval_batch_size: 256  #4096
loss_decimal_place: 4
metric_decimal_place: 4

**预期**
bert4rec所报结果：
test result[0m: {'recall@5': 0.0126, 'recall@10': 0.0232, 'recall@20': 0.0408, 'recall@50': 0.0769, 'mrr@5': 0.0061, 'mrr@10': 0.0075, 'mrr@20': 0.0087, 'mrr@50': 0.0098, 'ndcg@5': 0.0077, 'ndcg@10': 0.0111, 'ndcg@20': 0.0155, 'ndcg@50': 0.0227, 'precision@5': 0.0025, 'precision@10': 0.0023, 'precision@20': 0.002, 'precision@50': 0.0015}

BPR所报结果：
test result[0m: {'recall@5': 0.0226, 'recall@10': 0.0383, 'recall@20': 0.0598, 'recall@50': 0.0958, 'mrr@5': 0.0117, 'mrr@10': 0.0137, 'mrr@20': 0.0152, 'mrr@50': 0.0163, 'ndcg@5': 0.0144, 'ndcg@10': 0.0194, 'ndcg@20': 0.0248, 'ndcg@50': 0.0318, 'precision@5': 0.0045, 'precision@10': 0.0038, 'precision@20': 0.003, 'precision@50': 0.0019}

**实验环境（请补全下列信息）：**
 - 操作系统:  Windows
- RecBole 版本 0.2.1
 - Python 版本 3.83
- PyTorch 版本 1.10.0+cu113

**Comment by Sherry-XLL** [comment]
您好，@zyx1017！感谢您对 RecBole 的关注！

在 RecBole 中 BERT4Rec 属于序列推荐模型，默认的数据划分方式是留一法。而 BPR 是通用推荐模型，一般会按照比例对数据集进行划分，二者的运行结果并不能直接进行对比。

关于您提到的模型表现差，可能是由于没有对超参数进行搜索，或者数据集的稀疏程度影响了 BERT4Rec 的效果，需进一步分析实验细节。
